src/lib.rs

```diff
@@ -1399,6 +1399,10 @@ pub fn throw(s: &str) -> ! {
 /// function, unlike `panic!` on other platforms, **will not run destructors**.
 /// It's recommended to return a `Result` where possible to avoid the worry of
 /// leaks.
+///
+/// If you need destructors to run, consider using `panic!` with the
+/// `catch-unwind` feature enabled, which catches panics and converts them to
+/// JavaScript exceptions while properly running destructors.
```

Comment:
Actually this will run destructors when built with `-Cpanic=unwind`. Though it's unclear if this is the intended behavior by the Rust compiler. I don't imagine it will change.